### DISCUSSION TITLE
missing `autoPlay` in `LottieView`

### DISCUSSION THREAD

**Discussion by andyl-ioki** [lottie-react-native/lottie-react-native#916] [Q&A] [answered]
### Description

We updated `lottie-react-native` from `5.1.3` to `5.1.4`.
Our snapshot tests showed the following difference of a `LottieView`:
```jsx
<View
-   autoPlay={true}
    cacheComposition={true}
    collapsable={false}
    colorFilters={Array []}
    enableMergePathsAndroidForKitKatAndAbove={false}
    forwardedRef={[Function]}
    loop={true}
    onAnimationFinish={[Function]}
-   onLayout={[Function]}
    progress={0}
    resizeMode="contain"
    sourceJson="..."
    speed={1}
    style={...}
    textFiltersAndroid={Array []}
    textFiltersIOS={Array []}
+   useNativeLooping={false}
/>
```

### Comparison of `5.1.3` and `5.1.4`
So compared to `5.1.3` the `autoPlay` and `onLayout` is missing, but `useNativeLooping` was added.
I also checked your repo in regards of the changes of .4 and found [this](https://github.com/lottie-react-native/lottie-react-native/compare/v5.1.3...v5.1.4#diff-7000f7dd9109e1c8fdae7da17c10b0755dc4588ff8885bbfdc9b9f1cd28ed5d0R149) here:
```jsx
  render() {
    const { style, source, autoSize, autoPlay, ...rest } = this.props;  // autoPlay is not in rest anymore

    ...
    ...
    ...

    return (
      <View style={[aspectRatioStyle, sizeStyle, style]}>
        <AnimatedNativeLottieView
          ref={this.refRoot}
          {...rest}
          colorFilters={colorFilters}
          speed={speed}
          style={[aspectRatioStyle, sizeStyle || { width: '100%', height: '100%' }, style]}
          sourceName={sourceName}
          sourceJson={sourceJson}
          sourceURL={sourceURL}
          onAnimationFinish={this.onAnimationFinish}
        />
      </View>
    );
```

### Question
Do I get it right, that `autoPlay` was never used in `AnimatedNativeLottieView`? (Or it least now it is not used anymore, so passing it down does not make sense (anymore)).
So I don't have to worry to break anything with these changes above? 😅

**Comment by emilioicai** [comment, marked as answer]
not sure if I get your question,`autoPlay` is still an optional `LottieView` prop in 5.1.4 (https://github.com/lottie-react-native/lottie-react-native/blob/cc9cf5300fa9a394419ba81823e51057aa5e08f6/src/js/LottieView.js#L80). It will be passed to `AnimatedNativeLottieView` if provided but won't have any effect there as it is only used to trigger `play()` at `LottieView` level.

**Reply by andyl-ioki** [reply]
Hej @emilioicai,
thanks for your fast reply 👍 

> It will be passed to AnimatedNativeLottieView

As far as I understood the code, it won't get past down to `AnimatedNativeLottieView`, since:
```jsx
 const { style, source, autoSize, autoPlay, ...rest } = this.props;  // autoPlay is not in rest anymore
```
here `autoPlay` is extracted from props, so it is **not** in `rest` anymore.

Later you do not pass `autoPlay` explicitly to `AnimatedNativeLottieView` but `rest`:
```jsx
<AnimatedNativeLottieView
    ref={this.refRoot}
    {...rest}
    colorFilters={colorFilters}
    speed={speed}
    style={[aspectRatioStyle, sizeStyle || { width: '100%', height: '100%' }, style]}
    sourceName={sourceName}
    sourceJson={sourceJson}
    sourceURL={sourceURL}
    onAnimationFinish={this.onAnimationFinish}
/>
```

---

Anyway as you say, that it is not used in `AnimatedNativeLottieView`, it answered my initial question:
Is it a bug that `autoPlay` is missing in the rendered `View` of a `LottieView`?
Your answer was
> won't have any effect there as it is only used to trigger play() at LottieView level.
